go/arrow/array/struct.go

```diff
@@ -259,7 +259,7 @@ func (b *StructBuilder) Release() {
 }
 
 func (b *StructBuilder) Append(v bool) {
-	b.Reserve(1)
+	b.builder.reserve(1, b.resizeHelper)
```

Comment:
toss a small comment here so we don't forget why we did this in the future?

Reply:
Sure thing!